llvm/tools/llvm-mca/llvm-mca.cpp

```diff
@@ -225,10 +225,32 @@ static cl::opt<unsigned> StoreQueueSize("squeue",
                                         cl::desc("Size of the store queue"),
                                         cl::cat(ToolOptions), cl::init(0));
 
-static cl::opt<bool>
-    PrintInstructionTables("instruction-tables",
-                           cl::desc("Print instruction tables"),
-                           cl::cat(ToolOptions), cl::init(false));
+enum class InstructionTablesType { NONE, NORMAL, FULL };
+
+static cl::opt<enum InstructionTablesType> InstructionTablesOption(
+    "instruction-tables", cl::desc("Print instruction tables"),
+    cl::values(clEnumValN(InstructionTablesType::NONE, "none",
+                          "Do not print instruction tables"),
+               clEnumValN(InstructionTablesType::NORMAL, "normal",
+                          "Print instruction tables"),
+               clEnumValN(InstructionTablesType::NORMAL, "", ""),
+               clEnumValN(InstructionTablesType::FULL, "full",
+                          "Print instruction tables with additional"
+                          " information: bypass latency, LLVM opcode,"
+                          " used resources")),
+    cl::cat(ToolOptions), cl::init(InstructionTablesType::NONE),
+    cl::ValueOptional);
+
```

Comment:
nit: the naming of this sounds like it prints the instruction tables. rename to `shouldPrintInstructionTables`?